---
title: "DRD_FinalReport_Group1"
author: Amir Aynede, Kian Keshani, Bianca Gustini, Yuri Dobrokhotov, Favour Osese Otoijamun, Priscilla Castro Vargas
date: "2025-06-27"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

---
title: Comprehensive Analysis of DNA Methylation Using the Illumina HumanMethylation450 BeadChip Platform
output: html_document
date: "2025-06"
---

# Introduction

# Analytical pipeline and R code

## STEP 0. BASIC PREPARATION (Environment Setting and Importing Libraries)

# Set the proper working directory where the data is stored
```{r results="hide", message=FALSE, warning=FALSE}
setwd("~/chunche/DRD_2025_Project")
```

# Install and load the necessary packages
```{r results="hide", message=FALSE, warning=FALSE}
library(minfi)
library(minfiData)
library(IlluminaHumanMethylation450kmanifest)
library(IlluminaHumanMethylation450kanno.ilmn12.hg19)
library(shinyMethyl)
library(AnnotationDbi)
library(sva)
library(qqman)
library(gplots)
library(ggplot2)
install.packages("lib/SummarizedExperiment_1.38.0.tar.gz", repos = NULL)
```

## STEP 1. DATA IMPORT AND OBJECT CREATION
Load raw data with minfi and create an object called RGset storing the RGChannelSet object.

**Load the sample sheet**
```{r results="hide", message=FALSE, warning=FALSE}
SampleSheet <- read.csv("./data/raw/SampleSheet_Report_II.csv",header=T)
SampleSheet
```

**Load sample sheet using minfi's function**
```{r results="hide", message=FALSE, warning=FALSE}
baseDir <- ("./data/raw")
targets <- read.metharray.sheet(baseDir) 
targets
```

**Create RGChannelSet object**
```{r results="hide", message=FALSE, warning=FALSE}
RGset <- read.metharray.exp(targets = targets)
save(RGset,file="RGset_Report.RData")
load("RGset_Report.RData")
```

**Explore the RGset object**
```{r results="hide", message=FALSE, warning=FALSE}
RGset
str(RGset)
```

## STEP 2. EXTRACT RED AND GREEN CHANNELS
Create dataframes to store the red and green fluorescences.

**Extract fluorescence intensity info for Red and Green channels**
```{r results="hide", message=FALSE, warning=FALSE}
Red <- data.frame(getRed(RGset))
dim(Red)
head(Red)
Green <- data.frame(getGreen(RGset))
dim(Green)
head(Green)
```

## STEP 3. PROBE INFO AND ADDRESS CHECKS 
Determine whether address 18756452 belongs to a Type I or Type II probe by retrieving the red and green fluorescence values for this address and checking the manifest file.

**Explore manifest and probe information**
```{r results="hide", message=FALSE, warning=FALSE}
getManifest(RGset)
getManifestInfo(RGset)
getProbeInfo(RGset)
ProbeInfo_I <- data.frame(getProbeInfo(RGset))
dim(ProbeInfo_I)
head(getProbeInfo(RGset, type = "II"))
ProbeInfo_II <- data.frame(getProbeInfo(RGset, type = "II"))
dim(ProbeInfo_II)
```

**Check probe with address 18756452**
```{r results="hide", message=FALSE, warning=FALSE}
ProbeInfo_I[ProbeInfo_I$AddressA == "18756452",]
ProbeInfo_I[ProbeInfo_I$AddressB == "18756452",]
ProbeInfo_II[ProbeInfo_II$AddressB == "18756452",]
Probe_Info_18756452 <- ProbeInfo_II[ProbeInfo_II$AddressA == "18756452",]
```

**Extract Red/Green fluorescence for address 18756452**
```{r results="hide", message=FALSE, warning=FALSE}
Red_fluorescences <- Red[rownames(Red) == "18756452",]
Green_fluorescences <- Green[rownames(Green) == "18756452",]
```

++ FILL THE FOLLOWING TABLE.
++ ANSWER TO PROBE.

## STEP 4. Creation of the MSet.raw oject

**Extract methylated and unmethylated signals**
```{r results="hide", message=FALSE, warning=FALSE}
MSet.raw <- preprocessRaw(RGset)
MSet.raw
save(MSet.raw, file = "./data/processed/MSet_raw.RData")
Meth <- as.matrix(getMeth(MSet.raw))
str(Meth)
head(Meth)
Unmeth <- as.matrix(getUnmeth(MSet.raw))
str(Unmeth)
head(Unmeth)
```

**Check probe cg01523029 in MethylSet**
```{r results="hide", message=FALSE, warning=FALSE}
Unmeth[rownames(Unmeth) == "cg01523029",]
Meth[rownames(Meth) == "cg01523029",]

load("data/raw/Illumina450Manifest.RData")
Illumina450Manifest_clean <- Illumina450Manifest[Illumina450Manifest$CHR != "",]
Illumina450Manifest_clean <- droplevels(Illumina450Manifest_clean)
Probe_Info_cg01523029 <- Illumina450Manifest_clean[Illumina450Manifest_clean$IlmnID == "cg01523029",]
```

**Out-of-band signals: compare RGset and Methset for cg01523029**
```{r results="hide", message=FALSE, warning=FALSE}
Red[rownames(Red) == "18756452",]
Green[rownames(Green) == "18756452",]
Unmeth[rownames(Unmeth) == "cg01523029",]
Meth[rownames(Meth) == "cg01523029",]
```

## 5. Quality control
Perform quality checks and provide a brief comment to each step. 

### a. QCplot
```{r message=FALSE, warning=FALSE}
pdf("QCplot.pdf")
qc <- getQC(MSet.raw)
qc
plotQC(qc)
dev.off()
```
++COMMENT.

### b. Check the intensity of negative controls
```{r message=FALSE, warning=FALSE}
getProbeInfo(RGset, type = "Control")
NegativeControl <- data.frame(getProbeInfo(RGset, type = "Control"))
table(NegativeControl$Type)
#head(getProbeInfo(RGset, type = "Control")) - #ensuring RGset contains control probes
controlStripPlot(RGset, controls = "NEGATIVE")
```
++COMMENT.

### c. Calculation of detection p-values
```{r message=FALSE, warning=FALSE}
Detection_pvalue <- detectionP(RGset)
save(Detection_pvalue, file = "Detection_pvalue.RData")
Over_Threshold <- Detection_pvalue > 0.05
head(Over_Threshold)
table(Over_Threshold)
sum(Over_Threshold)
paste0(sprintf('There are %s failed probes with a detection p-value higher than 0.05', sum(Over_Threshold)))
Over_Threshold_Per_Sample <- colSums(Over_Threshold)
Over_Threshold_Per_Sample
summary(Over_Threshold)
```

++COMMENT.
++how many probes have a detection p-value higher than 0.05?


## STEP 6. Calculation of Beta and M values, Plot Densities
Calculate raw beta and M values and plot the densities of mean methylation values, dividing the samples in CTRL and DIS (suggestion: subset the beta and M values matrixes in order to retain CTRL or DIS subjects and apply the function mean to the 2 subsets). 

```{r message=FALSE, warning=FALSE}
beta <- getBeta(MSet.raw)
M <- getM(MSet.raw)
beta_df <- data.frame(beta)
M_df <- data.frame(M)

pheno <- read.csv("data/raw/SampleSheet_Report_II.csv", header = TRUE, stringsAsFactors = TRUE)
pheno$SampleID <- paste(pheno$Sentrix_ID, pheno$Sentrix_Position, sep = "_")

wt_samples <- pheno$SampleID[pheno$Group == "CTRL"]
mut_samples <- pheno$SampleID[pheno$Group == "DIS"]

beta_wt <- beta[, wt_samples]
density_beta_wt <- density(apply(beta_wt, 1, mean, na.rm = TRUE), na.rm = TRUE)

SampleSheet$Group
beta_ctrl <- beta_df[SampleSheet$Group == "CTRL",]
beta_dis <- beta_df[SampleSheet$Group == "DIS",]
M_ctrl <- M_df[SampleSheet$Group == "CTRL",]
M_dis <- M_df[SampleSheet$Group == "DIS",]

mean_of_beta_ctrl <- apply(beta_ctrl, 1, mean, na.rm = TRUE)
mean_of_beta_dis <- apply(beta_dis, 1, mean, na.rm = TRUE)
mean_of_M_ctrl <- apply(M_ctrl, 1, mean, na.rm = TRUE)
mean_of_M_dis <- apply(M_dis, 1, mean, na.rm = TRUE)
```

**Plotting the density of mean methylation values.**
```{r message=FALSE, warning=FALSE}
dev.new()
par(mfrow = c(1, 2))
plot(density(mean_of_beta_ctrl, na.rm = TRUE), main = "Density of Beta Values", col = "black")
lines(density(mean_of_beta_dis, na.rm = TRUE), col = "orange")

plot(density(mean_of_M_ctrl, na.rm = TRUE), main = "Density of M Values", col = "black")
lines(density(mean_of_M_dis, na.rm = TRUE), col = "orange")
```

++ Do you see any difference between the two groups?

## STEP 7. Data normalization with preprocessNoob
Normalize the data using preprocessNoob and compare raw data and normalized data. Produce a plot with 6 panels in which, for both raw and normalized data, you show the density plots of beta mean values according to the chemistry of the probes, the density plot of beta standard deviation values according to the chemistry of the probes and the boxplot of beta values. 

**Subset beta values by probe type**
```{r message=FALSE, warning=FALSE}
dfI <- Illumina450Manifest_clean[Illumina450Manifest_clean$Infinium_Design_Type=="I",]
dfI <- droplevels(dfI)
dfII <- Illumina450Manifest_clean[Illumina450Manifest_clean$Infinium_Design_Type=="II",]
dfII <- droplevels(dfII)

beta_I <- beta[rownames(beta) %in% dfI$IlmnID,]
beta_II <- beta[rownames(beta) %in% dfII$IlmnID,]

mean_of_beta_I <- apply(beta_I,1,mean)
mean_of_beta_II <- apply(beta_II,1,mean)
d_mean_of_beta_I <- density(mean_of_beta_I,na.rm=T)
d_mean_of_beta_II <- density(mean_of_beta_II,na.rm=T)
```

**Calculation of the densities of the standard deviations**
This can be calculated using the function sd():
```{r results="hide", message=FALSE, warning=FALSE}
sd_of_beta_I <- apply(beta_I, 1, sd, na.rm = TRUE)
sd_of_beta_II <- apply(beta_II, 1, sd, na.rm = TRUE)
d_sd_of_beta_I <- density(sd_of_beta_I,)
#d_sd_of_beta_II <- density(sd_of_beta_II) #doesn't work. We need to find out why- all na values should have been omited
d_sd_of_beta_II <- density(na.omit(sd_of_beta_II))
```

**Use of preprocessNoob**
```{r results="hide", message=FALSE, warning=FALSE}
?preprocessNoob
```

According to the help page, the input can be an RGset or MehylSet object, so proceed to load the RGset object.
```{r results="hide", message=FALSE, warning=FALSE}
#load("~/Dropbox/DRD_2025/3/RGset.RData")
#RGset
?GenomicRatioSet

preprocessNoob_results <- preprocessNoob(RGset)
beta_preprocessNoob <- getBeta(preprocessNoob_results)
save(beta_preprocessNoob, file = "beta_preprocessNoob.RData")
```

Divide normalized beta matrix by probe type and calculate statistics.
```{r results="hide", message=FALSE, warning=FALSE}
beta_preprocessNoob_I <- beta_preprocessNoob[rownames(beta_preprocessNoob) %in% dfI$IlmnID,]
beta_preprocessNoob_II <- beta_preprocessNoob[rownames(beta_preprocessNoob) %in% dfII$IlmnID,]
mean_of_beta_preprocessNoob_I <- apply(beta_preprocessNoob_I, 1, mean)
mean_of_beta_preprocessNoob_II <- apply(beta_preprocessNoob_II, 1, mean)
d_mean_of_beta_preprocessNoob_I <- density(mean_of_beta_preprocessNoob_I, na.rm = TRUE)
d_mean_of_beta_preprocessNoob_II <- density(mean_of_beta_preprocessNoob_II, na.rm = TRUE)
sd_of_beta_preprocessNoob_I <- apply(beta_preprocessNoob_I, 1, sd)
sd_of_beta_preprocessNoob_II <- apply(beta_preprocessNoob_II, 1, sd)
d_sd_of_beta_preprocessNoob_I <- density(sd_of_beta_preprocessNoob_I, na.rm = TRUE)
d_sd_of_beta_preprocessNoob_II <- density(sd_of_beta_preprocessNoob_II, na.rm = TRUE)
```

**Plot the data**
```{r results="hide", message=FALSE, warning=FALSE}
#pdf("Plot_comparison_raw_preprocessNoob.pdf", height = 7, width = 15)
#par(mfrow = c(2, 3))
plot(d_mean_of_beta_I, col = "blue", main = "raw beta")
lines(d_mean_of_beta_II, col = "red")
plot(d_sd_of_beta_I, col = "blue", main = "raw sd")
lines(d_sd_of_beta_II, col = "red")
boxplot(beta)

plot(d_mean_of_beta_preprocessNoob_I, col = "blue", main = "preprocessNoob beta")
lines(d_mean_of_beta_preprocessNoob_II, col = "red")
plot(d_sd_of_beta_preprocessNoob_I, col = "blue", main = "preprocessNoob sd")
lines(d_sd_of_beta_preprocessNoob_II, col = "red")
boxplot(beta_preprocessNoob)
```

**Improve the boxplot visualization by group**
```{r results="hide", message=FALSE, warning=FALSE}
("Plot_comparison_raw_preprocessNoob control vs disease.pdf", height = 7, width = 15)
par(mfrow = c(2, 3))
plot(d_mean_of_beta_I, col = "blue", main = "raw beta")
lines(d_mean_of_beta_II, col = "red")
plot(d_sd_of_beta_I, col = "blue", main = "raw sd")
lines(d_sd_of_beta_II, col = "red")
boxplot(beta,
        las = 2,
        col = c("green", "magenta")[pheno$Group],
        ylab = 'Beta values',
        xlab = 'Samples',
        main = 'Boxplot of Beta Values')

plot(d_mean_of_beta_preprocessNoob_I, col = "blue", main = "preprocessNoob beta")
lines(d_mean_of_beta_preprocessNoob_II, col = "red")
plot(d_sd_of_beta_preprocessNoob_I, col = "blue", main = "preprocessNoob sd")
lines(d_sd_of_beta_preprocessNoob_II, col = "red")
boxplot(beta_preprocessNoob,
        las = 2,
        col = c("green", "magenta")[pheno$Group],
        ylab = 'Beta values',
        xlab = 'Samples',
        main = 'Boxplot of Normalised Beta Values')
dev.off()
```

++Provide a short comment about the changes you observe.
++Do you think that the normalization approach that you used is appropriate considering this specific dataset? Try to color the boxplots according to the group (WT and MUT) and check whether the distribution of methylation values is different between the two groups, before and after normalization â€¦ 

## STEP 8. Principal Component Analysis
Perform a PCA on the matrix of normalized beta values generated in step 7. 

**Perform PCA on normalized beta values**
```{r results="hide", message=FALSE, warning=FALSE}
pca_results <- prcomp(t(beta_preprocessNoob),scale=T)
par(mfrow=c(1,1))
plot(pca_results,main="PCA results",ylim=c(0,150000))
pca_results$x
```

**Plot PCA by group, sex, and Sentrix_ID**
```{r results="hide", message=FALSE, warning=FALSE}
pheno$Group
palette(c('orange', 'green'))
plot(pca_results$x[, 1], pca_results$x[, 2], cex = 1, pch = 17, col = pheno$Group, main = 'PCA by group', xlab = "PC1", ylab = "PC2", xlim = c(-1000, 1000), ylim = c(-1000, 1000))
text(pca_results$x[, 1], pca_results$x[, 2], labels = (pheno$SampleID), cex = 0.7, pos = 1)
legend('bottomright', legend = levels(pheno$Group), col = c(1:nlevels(pheno$Group)), pch = 17)

pheno$Sex
pheno$Sex <- as.factor(pheno$Sex)
palette(c('pink', 'aquamarine'))
plot(pca_results$x[, 1], pca_results$x[, 2], cex = 2, pch = 17, col = pheno$Sex, main = 'PCA by sex', xlab = "PC1", ylab = "PC2", xlim = c(-1000, 1000), ylim = c(-1000, 1000))
text(pca_results$x[, 1], pca_results$x[, 2], labels = (pheno$SampleID), cex = 0.7, pos = 1)
legend('bottomright', legend = levels(pheno$Sex), col = c(1:nlevels(pheno$Sex)), pch = 17)

pheno$Sentrix_ID
pheno$Sentrix_ID <- as.factor(pheno$Sentrix_ID)
palette(c('red', 'blue', 'green', 'yellow', 'purple'))
plot(pca_results$x[, 1], pca_results$x[, 2], cex = 2, pch = 17, col = pheno$Sentrix_ID, main = 'PCA by Sentrix_ID', xlab = "PC1", ylab = "PC2", xlim = c(-1000, 1000), ylim = c(-1000, 1000))
text(pca_results$x[, 1], pca_results$x[, 2], labels = (pheno$SampleID), cex = 0.7, pos = 1)
legend('bottomright', legend = levels(pheno$Sentrix_ID), col = c(1:nlevels(pheno$Sentrix_ID)), pch = 17)
```

++Comment the plot.
++Do you see any outlier? Do the samples divide according to the group? Do they divide according to the sex of the samples? Do they divide according to the batch, that is the column Sentrix_ID?). 

**Additional PCA visualizations**
```{r results="hide", message=FALSE, warning=FALSE}
pca <- prcomp(beta_preprocessNoob, scale = TRUE)
par(mfrow = c(1, 1))
plot(pca$x[, 1], pca$x[, 2], cex = 2, pch = 18, xlab = "PC1", ylab = "PC2", xlim = c(-500, 800), main = "PCA")
text(pca$x[, 1], pca$x[, 2], labels = rownames(pca$x), pos = 4, cex = 0.5)

pcr <- data.frame(pca$rotation[, 1:3], Group = targets$Group)
ggplot(pcr, aes(PC1, PC2, color = Group)) + geom_point(size = 4)
pcr <- data.frame(pca$rotation[, 1:3], Group = targets$Sex)
ggplot(pcr, aes(PC1, PC2, color = Group)) + geom_point(size = 4)
```


## STEP 9. Differential Methylation Analysis
Using the matrix of normalized beta values generated in step 7, identify differentially methylated probes between CTRL and DIS groups using a t-test.

**Example for a single probe**
```{r results="hide", message=FALSE, warning=FALSE}
t_test <- t.test(beta_preprocessNoob[1, ] ~ pheno$Group)
t_test
t_test$p.value
```

**Define a function to apply test to all probes**
```{r results="hide", message=FALSE, warning=FALSE}
My_ttest_function <- function(x) {
  t_test <- t.test(x ~ pheno$Group)
  return(t_test$p.value)
}
```

**Apply the function to all probes**
```{r results="hide", message=FALSE, warning=FALSE}
pValues_ttest <- apply(beta_preprocessNoob, 1, My_ttest_function)
```

**Create data.frame with beta values and p-values**
```{r results="hide", message=FALSE, warning=FALSE}
final_ttest <- data.frame(beta_preprocessNoob, pValues_ttest)
```

**Order probes by p-value**
```{r results="hide", message=FALSE, warning=FALSE}
final_ttest <- final_ttest[order(final_ttest$pValues_ttest), ]
head(final_ttest)
```

**Filter for significant probes (p < 0.05)**
```{r results="hide", message=FALSE, warning=FALSE}
significant_ttest <- final_ttest[final_ttest$pValues_ttest <= 0.05, ]
```

## STEP 10. Multiple Test Correction

**Apply multiple test correction and set a significant threshold of 0.05**
```{r results="hide", message=FALSE, warning=FALSE}
corrected_pValues_BH <- p.adjust(final_ttest$pValues_ttest, "BH")
corrected_pValues_Bonf <- p.adjust(final_ttest$pValues_ttest, "bonferroni")
final_ttest_corrected <- data.frame(final_ttest, corrected_pValues_BH, corrected_pValues_Bonf)
head(final_ttest_corrected)
```

**Visualize distributions of p-values and corrected p-values**
```{r results="hide", message=FALSE, warning=FALSE}
colnames(final_ttest_20k_corrected)
boxplot(final_ttest_20k_corrected[,9:11])
```

**Determining the amount of probes that remain after the MTC**
```{r results="hide", message=FALSE, warning=FALSE}
dim(final_ttest_corrected[final_ttest_corrected$pValues_ttest <= 0.05, ])
dim(final_ttest_corrected[final_ttest_corrected$corrected_pValues_BH <= 0.05, ])
dim(final_ttest_corrected[final_ttest_corrected$corrected_pValues_Bonf <= 0.05, ])
```

++How many probes do you identify as differentially methylated considering nominal pValues? How many after Bonferroni correction? How many after BH correction?

## STEP 11. Graphic the results of the differential methylation analysis.
Produce a volcano plot and a Manhattan plot of the results of differential methylation analysis. 

**a. Volcano plot**

Creation of the Volcano plot.
```{r results="hide", message=FALSE, warning=FALSE}
beta_first <- final_ttest_corrected[, 1:8]
beta_first_ctrl <- beta_first[, pheno$Group == "CTRL"]
mean_beta_first_ctrl <- apply(beta_first_ctrl, 1, mean)
beta_first_dis <- beta_first[, pheno$Group == "DIS"]
mean_beta_first_dis <- apply(beta_first_dis, 1, mean)
delta_first <- mean_beta_first_dis - mean_beta_first_ctrl
toVolcPlot <- data.frame(delta_first, -log10(final_ttest_corrected$pValues_ttest))
plot(toVolcPlot[, 1], toVolcPlot[, 2])
```

Highlight the probes with a nominal pValue<0.01 and a delta > 0.1.
```{r results="hide", message=FALSE, warning=FALSE}
pdf("Volcano_plot.pdf")
plot(toVolcPlot[, 1], toVolcPlot[, 2], pch = 16, cex = 0.5, main = 'volcano plot')
toHighlight <- toVolcPlot[toVolcPlot[, 1] > 0.1 & toVolcPlot[, 2] > (-log10(0.05)), ]
points(toHighlight[, 1], toHighlight[, 2], pch = 16, cex = 0.7, col = "red")
dev.off()
```

**b. Manhattan plot**

Annotate dataframe with genome annotation for each CpG probe.
```{r results="hide", message=FALSE, warning=FALSE}
load('~/Dropbox/DRD_2025/2/Illumina450Manifest_clean.RData')
final_ttest_corrected <- data.frame(rownames(final_ttest_corrected), final_ttest_corrected)
colnames(final_ttest_corrected)[1] <- "IlmnID"
final_ttest_corrected_annotated <- merge(final_ttest_corrected, Illumina450Manifest_clean, by = "IlmnID")
input_Manhattan <- final_ttest_corrected_annotated[colnames(final_ttest_corrected_annotated) %in% c("IlmnID", "CHR", "MAPINFO", "pValues_ttest")]
order_chr <- c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "X", "Y")
input_Manhattan$CHR <- factor(input_Manhattan$CHR, levels = order_chr)
input_Manhattan$CHR <- as.numeric(input_Manhattan$CHR)
manhattan(input_Manhattan, snp = "IlmnID", chr = "CHR", bp = "MAPINFO", p = "pValues_ttest")
pdf("Manhattan_plot.pdf")
manhattan(input_Manhattan, snp = "IlmnID", chr = "CHR", bp = "MAPINFO", p = "pValues_ttest", annotatePval = 0.00001, col = rainbow(24))
dev.off()
```


## 12. Heatmap
Produce an heatmap of the top 100 significant, differentially methylated probes.

```{r results="hide", message=FALSE, warning=FALSE}
colorbar <- c("green", "orange", "orange", "green", "orange", "orange", "green", "green")
input_heatmap = as.matrix(final_ttest[1:100, 1:8])
```

Complete linkage (default).
```{r results="hide", message=FALSE, warning=FALSE}
heatmap.2(input_heatmap, col = terrain.colors(100), Rowv = TRUE, Colv = TRUE, dendrogram = "both", key = TRUE, ColSideColors = colorbar, density.info = "none", trace = "none", scale = "none", symm = FALSE, main = "Complete linkage")
```

Single linkage.
```{r results="hide", message=FALSE, warning=FALSE}
heatmap.2(input_heatmap, col = terrain.colors(100), Rowv = TRUE, Colv = TRUE, hclustfun = function(x) hclust(x, method = 'single'), dendrogram = "both", key = TRUE, ColSideColors = colorbar, density.info = "none", trace = "none", scale = "none", symm = FALSE, main = "Single linkage")
```

Average linkage.
```{r results="hide", message=FALSE, warning=FALSE}
heatmap.2(input_heatmap, col = terrain.colors(100), Rowv = TRUE, Colv = TRUE, hclustfun = function(x) hclust(x, method = 'average'), dendrogram = "both", key = TRUE, ColSideColors = colorbar, density.info = "none", trace = "none", scale = "none", symm = FALSE, main = "Average linkage")
```

Custom of the colors.
```{r results="hide", message=FALSE, warning=FALSE}
col2 = colorRampPalette(c("green", "black", "red"))(100)
heatmap.2(input_heatmap, col = col2, Rowv = TRUE, Colv = TRUE, dendrogram = "both", key = TRUE, ColSideColors = colorbar, density.info = "none", trace = "none", scale = "none", symm = FALSE)
```
